examples/material_gallery/lib/demo/weathers_demo.dart

```diff
@@ -0,0 +1,495 @@
+
+  Widget build(BuildContext context) {
+    if (!assetsLoaded)
+      return new Text("Loading...");
+
+    return new Scaffold(
+      toolBar: new ToolBar(
+        center: new Text("Weather")
+      ),
+      body: new Material(
+        child: new Stack(
+          children: [
+            new SpriteWidget(weatherWorld),
+            new Align(
+              alignment: new FractionalOffset(0.5, 0.8),
+              child: new Row(
+                justifyContent: FlexJustifyContent.center,
+                children: [
+                  new WeatherButton(
+                    onPressed: () {
+                      setState(() {
+                        weatherWorld.weatherType = WeatherType.sun;
+                      });
+                    },
+                    selected: weatherWorld.weatherType == WeatherType.sun,
+                    icon: "packages/flutter_gallery_assets/icon-sun.png"
+                  ),
+                  new WeatherButton(
+                    onPressed: () {
+                      setState(() {
+                        weatherWorld.weatherType = WeatherType.rain;
+                      });
+                    },
+                    selected: weatherWorld.weatherType == WeatherType.rain,
+                    icon: "packages/flutter_gallery_assets/icon-rain.png"
+                  ),
+                  new WeatherButton(
+                    onPressed: () {
+                      setState(() {
+                        weatherWorld.weatherType = WeatherType.snow;
+                      });
+                    },
+                    selected: weatherWorld.weatherType == WeatherType.snow,
+                    icon: "packages/flutter_gallery_assets/icon-snow.png"
+                  )
+                ]
+              )
+            )
+          ]
+        )
+      )
+    );
+  }
+}
+
+const double _kWeatherButtonSize = 56.0;
+const double _kWeatherIconSize = 36.0;
+
+class WeatherButton extends StatelessComponent {
+  WeatherButton({this.icon, this.selected, this.onPressed});
+
+  final String icon;
+  final bool selected;
+  VoidCallback onPressed;
+
+  Widget build(BuildContext context) {
+    Color color;
+    if (selected)
+      color = Theme.of(context).primaryColor;
+    else
+      color = new Color(0x33000000);
+
+    return new Padding(
+      padding: new EdgeDims.all(15.0),
+      child: new Material(
+        color: color,
+        type: MaterialType.circle,
+        elevation: 0,
+        child: new Container(
+          width: _kWeatherButtonSize,
+          height: _kWeatherButtonSize,
+          child: new InkWell(
+            onTap: onPressed,
+            child: new Center(
+              child: new AssetImage(
+                name: icon,
+                width: _kWeatherIconSize,
+                height: _kWeatherIconSize
+              )
+            )
+          )
+        )
+      )
+    );
+  }
+}
+
+List<Color> _kBackgroundColorsTop = [
+  new Color(0xff5ebbd5),
```

Comment:
that is, the _kBackgroundColorsTop should be const, along with everything inside it.

Also, `<Color>[...]`.